src/backend/columnar/columnar_tableam.c

```diff
@@ -341,24 +365,33 @@ columnar_parallelscan_reinitialize(Relation rel, ParallelTableScanDesc pscan)
 static IndexFetchTableData *
 columnar_index_fetch_begin(Relation rel)
 {
-	ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-					errmsg("indexes not supported for columnar tables")));
+	Oid relfilenode = rel->rd_node.relNode;
+	if (PendingWritesInUpperTransactions(relfilenode, GetCurrentSubTransactionId()))
+	{
+		elog(ERROR, "cannot read from index when there is unflushed data in "
+					"upper transactions");
```

Comment:
Why can't we flush first and then proceed? No need to do this now, but maybe add an XXX comment so we can fix it later.

Reply:
let me add XXX to there